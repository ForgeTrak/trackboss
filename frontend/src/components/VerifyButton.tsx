import { Button } from '@chakra-ui/react';
import React, { useState } from 'react';

interface buttonProps {
    verified: boolean,
}

export default function VerifyButton(props: buttonProps) {
    const [verified, setVerified] = useState(props.verified);

    const handleClick = () => {
        setVerified(!verified);
    };

    return (
        <Button
            size="lg"
            variant={verified ? 'verified' : 'unverified'}
            onClick={handleClick}
        >
            {verified ? 'Verified' : 'Verify'}
        </Button>
    );
}
