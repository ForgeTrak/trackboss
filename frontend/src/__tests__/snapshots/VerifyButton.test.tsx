import React from 'react';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';
import VerifyButton from '../../components/VerifyButton';

describe('VerifyButton component', () => {
    it('renders correctly', () => {
        let button: ReactTestRenderer;
        act(() => {
            button = renderer.create(<VerifyButton verified={true}/>);
        });
        
        expect(button!.toJSON()).toMatchSnapshot();
    });
});
