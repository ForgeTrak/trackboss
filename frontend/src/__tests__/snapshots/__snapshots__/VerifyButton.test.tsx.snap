// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyButton component renders correctly 1`] = `
<button
  className="chakra-button css-5c2176"
  onClick={[Function]}
  type="button"
>
  Verified
</button>
`;
